x-pack/plugins/fleet/server/services/agents/crud.ts

```diff
@@ -59,7 +58,35 @@ export function removeSOAttributes(kuery: string) {
   return kuery.replace(/attributes\./g, '').replace(/fleet-agents\./g, '');
 }
 
-export async function listAgents(
+export type GetAgentsOptions =
+  | {
+      agentIds: string[];
+    }
+  | {
+      kuery: string;
+      showInactive?: boolean;
+    };
+
+export async function getAgents(esClient: ElasticsearchClient, options: GetAgentsOptions) {
+  let initialResults = [];
+
+  if ('agentIds' in options) {
+    initialResults = await getAgentsById(esClient, options.agentIds);
+  } else if ('kuery' in options) {
+    initialResults = (
+      await getAllAgentsByKuery(esClient, {
+        kuery: options.kuery,
+        showInactive: options.showInactive ?? false,
+      })
+    ).agents;
+  } else {
+    throw new IngestManagerError('Cannot get agents');
```

Comment:
Can surely improve this. Not sure what it should say. Something about required options?